kasa/aestransport.py

```diff
@@ -223,7 +225,8 @@ async def try_login(self, login_params: Dict[str, Any]) -> None:
 
         resp_dict = await self.send_secure_passthrough(request)
         self._handle_response_error_code(resp_dict, "Error logging in")
-        self._login_token = resp_dict["result"]["token"]
+        login_token = resp_dict["result"]["token"]
+        self._token_url = self._app_url.with_query(f"token={login_token}")
```

Comment:
Is this a typing issue, as the `self._token_url` is defined as optional str, but `with_query` returns a `URL` object?

Reply:
Yes it is.  I'm surprised mypy didn't complain

fixed in 4efaaa0